specs/corla.pvs

```diff
@@ -0,0 +1,923 @@
+% authentication systems mandate the use of multiple credentials.
+  credential: TYPE 
+  username: TYPE FROM credential
+  password: TYPE FROM credential
+% @trace Task order 1, page 4, "Uptime/Security"
+  complex_enough?: [password -> bool]
+% complex_check_update: [complex_enough? -> ?]
+  biometric: TYPE FROM credential
+  physical_token: TYPE FROM credential
+
+% @description One has authentication if a person has the appropriate
+% combination of identity and set of credentials.
+  authentication: TYPE FROM [person, set[credential]]
+% @description Two-factor authentication is a currently-popular form
+% of authentication that requires more than just knowledge (say, of a
+% username and password), and at least one physical mechanism, such as
+% a biometric or a physical token.
+  two_factor_authentication: TYPE =
+    [[username, password], [physical_token + biometric]]
+
+%%% In order to obtain a new, valid credential, some authority must
+%%% issue credentials to a specific person.
+  issue_credential: TYPE =
+    [cdos, state_election, person -> [person, two_factor_authentication]]
+
+%%% What are the credentials that have been issued to this person?
+  credential?: [person -> two_factor_authentication]
+
+%%% Is this person authenticated?
+  authenticated: [person, two_factor_authentication -> bool]
+
+%%% What follows are several features used to authenticate various
+%%% roles to their respective dashboards.
+
+%%% Each dashboard has an abstract state machine that captures the
+%%% dashboard's workflow and consequntly identifies which features are
+%%% visible to its users at various stages. Those abstract state
+%%% machines and features are modeled in the dashboard modules
+%%% below. We focus only on authentication here.
+
+% @todo kiniry Introduce an authentication monad and a state monad for
+% our ASMs.
+
+% @description Authenticate this county administrator to that county dashboard.
+  authenticate_county_administrator:
+    [county_dashboard, county_administrator, two_factor_authentication -> bool]
+% @description Authenticate this state administrator to that state-wide dashboard.
+  authenticate_state_administrator:
+    [state_wide_dashboard, state_administrator, two_factor_authentication -> bool]
+END authentication
+
+%%% The user interfaces of the system are the visible interactive
+%%% parts of an application. There are several different user
+%%% interfaces in the RLA Tool. The precise names of these interfaces
+%%% is still under discussion, but basically one is for CDOS personnel
+%%% responsible for the audit at the state-level, one is for county
+%%% personnel at the county-level, and one is for audit board members.
+
+user_interface: DATATYPE
+BEGIN
+  uploading_interface?: uploading_interface
+  cvr_uploading_interface?: cvr_uploading_interface
+  county_auditing_interface?: county_auditing_interface
+  audit_adjudication_interface?: audit_adjudication_interface
+END user_interface
```

Comment:
Added, and I refined the datatype appropriately.  See 2b9ce8902f8f5745f145830f9e2ef34417fdd9f3.